### PR TITLE
Add support for neg to NestedTensor

### LINKED ISSUE #86889: Support negation/subtraction of nested_tensor
### 🚀 The feature, motivation and pitch

These operations are not currently supported. Since `add` is supported, the implementation might not be too much work.

```
>>> - torch.nested.nested_tensor([torch.tensor([-2.3, 0.4, 5.3]), torch.tensor([-2.3, 0.4, 5.3])])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NotImplementedError: Could not run 'aten::neg' with arguments from the 'NestedTensorCPU' backend. This could be because the operator doesn't exist for this backend, or was omitted during the selective/custom build process (if using custom build). If you are a Facebook employee using PyTorch on mobile, please visit https://fburl.com/ptmfixes for possible resolutions. 'aten::neg' is only available for these backends: [CPU, CUDA, HIP, MPS, IPU, XPU, HPU, VE, Meta, PrivateUse1, PrivateUse2, PrivateUse3, FPGA, ORT, Vulkan, Metal, QuantizedCPU, QuantizedCUDA, QuantizedHIP, QuantizedMPS, QuantizedIPU, QuantizedXPU, QuantizedHPU, QuantizedVE, QuantizedMeta, QuantizedPrivateUse1, QuantizedPrivateUse2, QuantizedPrivateUse3, CustomRNGKeyId, MkldnnCPU, SparseCPU, SparseCsrCPU, SparseCsrCUDA, BackendSelect, Python, FuncTorchDynamicLayerBackMode, Functionalize, Named, Conjugate, Negative, ZeroTensor, ADInplaceOrView, AutogradOther, AutogradCPU, AutogradCUDA, AutogradHIP, AutogradXLA, AutogradMPS, AutogradIPU, AutogradXPU, AutogradHPU, AutogradVE, AutogradLazy, AutogradMeta, AutogradPrivateUse1, AutogradPrivateUse2, AutogradPrivateUse3, AutogradNestedTensor, Tracer, AutocastCPU, AutocastCUDA, FuncTorchBatched, FuncTorchVmapMode, Batched, VmapMode, FuncTorchGradWrapper, PythonTLSSnapshot, FuncTorchDynamicLayerFrontMode, PythonDispatcher].
```

### Alternatives

Depending on the situation, it may be possible to negate all values before putting them into the nested_tensor. Or else unbind() could be used and work inner tensor at a time.

### Additional context

_No response_

cc @cpuhrsch @jbschlosser @bhosmer @drisspg @mikaylagawarecki

### PR DESCRIPTION
**Pull request by cpuhrsch**
Partially fixes #86889

### CODE REVIEW

File: test/test_nestedtensor.py [hunk 1]
```diff
@@ -762,6 +748,20 @@ def test_nested_tensor_indexing(self, device, dtype):
...
 
+    @parametrize("func", [torch.nn.functional.relu,
+                          torch.nn.functional.relu_,
+                          torch.nn.functional.gelu,
+                          torch._C._nn.gelu_,
+                          torch.tanh,
+                          torch.tanh_,
+                          torch.neg])
+    def test_activations(self, device, func):
+        nt = random_nt(device, torch.float32, 3, (4, 4))
```
Comment (drisspg) [inline test/test_nestedtensor.py:759]:
```suggestion
        random_nt_noncontiguous_pair
```
Should we test both contiguous and noncontiguous with this func?

**Review by drisspg — APPROVED** [review]

File: test/test_nestedtensor.py (hunk 1 above)
Comment (cpuhrsch) [reply test/test_nestedtensor.py:759]: Unary ops currently only support contiguous inputs.

Comment (cpuhrsch) [reply test/test_nestedtensor.py:759]: I'll add an assertRaises

**Comment by cpuhrsch** [comment]
This reason this was reverted is the formatting of the names of the tests...

**Pull request closed without merging**